src/java.base/share/classes/sun/security/x509/AlgorithmId.java

```diff
@@ -108,9 +109,8 @@ public AlgorithmId(ObjectIdentifier oid, AlgorithmParameters algparams) {
             try {
                 encodedParams = algParams.getEncoded();
             } catch (IOException ioe) {
-                // Ignore this at the moment. This exception can occur
-                // if AlgorithmParameters was not initialized yet. Will
-                // try to re-getEncoded() again later.
+                throw new IllegalStateException(
+                        "AlgorithmParameters not initialized", ioe);
```

Comment:
Did you have a chance to look at the caller and make sure this behavioral update is safe?

Reply:
Would it be more correct to change the message to "AlgorithmParameters are not initialized or cannot be decoded"?